components/inputs/input-date-range.js

```diff
@@ -95,18 +103,21 @@ class InputDateRange extends RtlMixin(LocalizeCoreElement(LitElement)) {
 		if (!this.label) {
 			console.warn('d2l-input-date-range component requires label text');
 		}
+
+		this._endLabel = this.endLabel ? this.endLabel : this.localize('components.input-date-range.endDate');
+		this._startLabel = this.startLabel ? this.startLabel : this.localize('components.input-date-range.startDate');
```

Comment:
Technically I think these may need to be re-computed in `updated` since any time `end-label` or `start-label` are set you'd need to re-compute. Instead of properties, I think it might be easier to just create private getters for these and then call them from `render()` and `validationMessageBadInput()`.